docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,752 @@
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Specify Backend"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "outputs": [],
+   "source": [
+    "# declare quantum instance\n",
+    "backend = Aer.get_backend('aer_simulator')\n",
+    "qi = QuantumInstance(backend, shots = batch_size)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Definition of quantum generator and the respective gradient"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def generator_(qnn: QuantumCircuit,\n",
+    "               parameters: Union[ParameterVector, ParameterExpression, List[ParameterExpression]]) -> TorchConnector:\n",
+    "    \"\"\"\n",
+    "    Args:\n",
+    "        qnn: Quantum neural network ansatz given as a quantum circuit.\n",
+    "        parameters: The parameters of the quantum neural network which are trained.\n",
+    "    Returns:\n",
+    "        Quantum neural network compatible with PyTorch\n",
+    "    \"\"\"\n",
+    "    circuit_qnn = CircuitQNN(qnn, input_params=[], weight_params = parameters,\n",
+    "                             quantum_instance=qi, sampling=True, sparse=False,\n",
+    "                             input_gradients=True, interpret=lambda x: grid_elements[x])\n",
+    "    # We use the Qiskit TorchConnector to ensure compatibility with PyTorch\n",
+    "    return TorchConnector(circuit_qnn)\n",
+    "\n",
+    "def generator_grad(qnn: QuantumCircuit,\n",
+    "                   parameters: Union[ParameterVector, ParameterExpression, List[ParameterExpression]],\n",
+    "                   param_values: Iterable,\n",
+    "                   grad_method: str = 'param_shift') -> Iterable:\n",
+    "    \"\"\"\n",
+    "    Custom generator gradient\n",
+    "    Args:\n",
+    "        qnn: Quantum neural network ansatz given as a quantum circuit.\n",
+    "        parameters: The parameters of the quantum neural network which are trained.\n",
+    "        param_values: The current values of the quantum neural network parameters.\n",
+    "        grad_method: Method used to compute the gradients {'param_shift', 'lin_comb', 'fin_diff'}\n",
+    "    Returns:\n",
+    "        List of gradients for the sampling probabilities of the quantum neural network.\n",
+    "    \"\"\"\n",
+    "    grad = Gradient(grad_method=grad_method).gradient_wrapper(StateFn(qnn), parameters, backend=qi)\n",
+    "    grad_values = grad(param_values)\n",
+    "    return grad_values.tolist()"
+   ]
+  },
```

Comment:
I am not convinced about having these function definitions here.... specially because they are only used once in the code, and they are a couple of lines each. I would say, either we create some kind of `qgan_utils.py` file to import these functions from (if we really want to have them for some reason), or we just remove them and instantiate the QNN and gradient directly later on.

Reply:
Oh, I see. 
Well, we do want to have this tutorial basically to replace the existing qGAN code. Hence, we still need to define the generator somewhere for illustrative (as well as practical) reasons.